### PR TITLE
ci: testing bad behavior change

### PR DESCRIPTION
**Pull request by suztomo**
No need to review.

This PR contains a behavior change that would break other libraries.

### CODE REVIEW

File: google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java [hunk 1]
```diff
@@ -876,7 +876,7 @@ public HttpResponse execute() throws IOException {
       executeException = null;
 
       // run the interceptor
-      if (executeInterceptor != null) {
+      if (executeInterceptor == null) { // This code is wrong. Just for testing.
         executeInterceptor.intercept(this);
```
Comment (suztomo) [inline google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java:880]: google-cloud-storage's test threw `NullPointerException` correctly.

Screen Shot 2021-05-05 at 17 00 57

Comment (suztomo) [reply google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java:880]: However this approach does not work for Gradle projects such as "LTS Library Test / lts-library-test (beam)".

File: .github/workflows/lts-test.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+           -Dclirr.skip=true \
+           -Denforcer.skip=true \
+           -Dmaven.javadoc.skip=true \
+           -Dgcloud.download.skip=true
+      # The lts_tests branch has not merged to master but it hosts GitHub Actions.
+      - uses: GoogleCloudPlatform/cloud-opensource-java/lts-test@lts_tests
+        with:
+          repository: ${{ matrix.repository }}
+          version_override_key: http.version
+          version_override_value: 1.39.2
```
Comment (suztomo) [inline .github/workflows/lts-test.yaml:45]: This should not be manually set.

File: .github/workflows/lts-test.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          - java-storage
+          - java-trace
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: Install this repository's artifacts to local Maven repository
+        run: |
+          mvn  --batch-mode --show-version --no-transfer-progress install \
```
Comment (suztomo) [inline .github/workflows/lts-test.yaml:34]: Is it possible to specify some strange version number (that would not conflict) when installing this artifact?

Comment (suztomo) [reply .github/workflows/lts-test.yaml:34]: Does `{x-version-update:google-http-client-parent:current}` help?

**Comment by suztomo** [comment]
Why does "LTS Library Test / lts-library-test (beam)" succeed?

**Comment by suztomo** [comment]
Maybe the Gradle is not reading the local Maven repository first?

**Comment by suztomo** [comment]
Version `1.39.2-sp.2` didn't work for the assembly module in GitHub Actions (it works in my MacBook):

```
[INFO] --- maven-assembly-plugin:3.3.0:single (make-assembly) @ google-http-client-assembly ---
[INFO] Reading assembly descriptor: assembly.xml
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Parent for the Google HTTP Client Library for Java 1.39.2-sp.2 SUCCESS [ 10.980 s]
[INFO] Google HTTP Client Library for Java 1.39.2-sp.2 .... SUCCESS [ 19.736 s]
[INFO] Protocol Buffer extensions to the Google HTTP Client Library for Java. 1.39.2-sp.2 SUCCESS [  6.046 s]
[INFO] Assembly for the Google HTTP Client Library for Java 1.39.2-sp.2 FAILURE [  6.234 s]
[INFO] Google App Engine extensions to the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] Android Platform Extensions to the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] Apache HTTP transport v2 for the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] GSON extensions to the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] Jackson 2 extensions to the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] XML extensions to the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] Google APIs Client Library Findbugs custom plugin. 1.39.2-sp.2 SKIPPED
[INFO] Shared classes used for testing of artifacts in the Google HTTP Client Library for Java. 1.39.2-sp.2 SKIPPED
[INFO] Simple example for the Dailymotion API. 1.39.2-sp.2  SKIPPED
[INFO] Google HTTP Client Library for Java BOM 1.39.2 ..... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  44.027 s
[INFO] Finished at: 2021-05-07T14:27:48Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single (make-assembly) on project google-http-client-assembly: Failed to create assembly: Error adding file to archive: /home/runner/work/google-http-java-client/google-http-java-client/google-http-client-assembly/../google-http-client-android/target/site/dependencies.html -> [Help 1]
```

**Comment by suztomo** [comment]
Now Beam test fails as expected due to the null check. Good.

Screen Shot 2021-05-07 at 12 20 13

File: .github/workflows/lts-test.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - id: find_library_version
+        name: Find the library version in this branch
+        run: |
+          sudo apt-get install libxml2-utils
+          VERSION=`sed -e 's/xmlns="[^\s\\]*"//' pom.xml | xmllint --xpath '/project/version/text()' -`
+          echo "::set-output name=version::${VERSION}"
```
Comment (suztomo) [inline .github/workflows/lts-test.yaml:37]: @elharo (a followup from yesterday's discussion) This command extracts the version from the root pom.xml. This configuration file no longer needs to specify a hard-coded version. The value is then used at the end of this file as `version_override_value` parameter.

(I tweaked the command from https://github.com/googleapis/java-cloud-bom/blob/8a45813a6590d3a25dcaad4fa5d6beaeb341b3d3/.kokoro/bom-upper-bounds-check.sh#L43)

This sed+xmllint approach should work for Yoshi repositories. Other projects, such as [grpc-java](https://github.com/grpc/grpc-java/blob/master/build.gradle#L21) and [Beam](https://github.com/apache/beam/blob/master/gradle.properties#L27), will have other regular expression to extract their versions.

**Comment by elharo** [comment]
can this be closed now?

**Comment by suztomo** [comment]
Yes. Closing. Thank you for followup.

**Pull request closed without merging**